v0/destinations/candu/transform.js

```diff
@@ -0,0 +1,97 @@
+const { EventType } = require("../../../constants");
+const {
+  CustomError,
+  defaultRequestConfig,
+  constructPayload,
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  removeUndefinedAndNullAndEmptyValues
+} = require("../../util");
+const { endpoint, identifyDataMapping, trackDataMapping } = require("./config");
+
+const responseBuilder = (body, { Config }) => {
+  const payload = body;
+  payload.context = {
+    source: "RudderStack"
+  };
+  const response = defaultRequestConfig();
+  response.endpoint = endpoint;
+  response.body.JSON = removeUndefinedAndNullAndEmptyValues(payload);
+  if (!Config.apiKey.trim())
+    throw new CustomError(`[CANDU]:: apiKey cannot be empty.`, 400);
```

Comment:
As this is required by all the message types, this should be part of the primary validations (along with the `message.type` check)